src/main/java/com/bootcamp/demo/service/LoginService.java

```diff
@@ -0,0 +1,72 @@
+package com.bootcamp.demo.service;
+
+import com.bootcamp.demo.model.Session;
+import com.bootcamp.demo.model.User;
+import com.bootcamp.demo.repository.SessionRepository;
+import com.bootcamp.demo.repository.UserRepository;
+import org.springframework.stereotype.Service;
+
+import java.time.Instant;
+
+/**
+ * LoginService implementation
+ * Next to add: forgotPassword()
+ */
+@Service
+public class LoginService implements ILoginService {
+    private UserRepository userRepository;
+    private SessionRepository sessionRepository;
+
+    public LoginService(UserRepository userRepository, SessionRepository sessionRepository) {
+        this.userRepository = userRepository;
+        this.sessionRepository = sessionRepository;
+    }
+
+    /**
+     * @param new_user the new user
+     * @return true, if user was successfully saved, or false if email is already used
+     */
+    @Override
+    public boolean registerUser(User new_user) {
+        if(userRepository.findByEmail(new_user.getEmail())!= null)
+            return false;
+        return userRepository.save(new_user);
```

Comment:
Some coding guidelines:
* the file needs to be properly formatted
* apply the standard java naming convention: newUser for the method parameter
* always use accolades {} for an if statement, even if it has a single step executed in case it's true
* in general, we want to avoid multiple return statements for a method - this increases readability and we avoid potential errors in the flow. in this example, we could have written:
```
    public boolean registerUser(User newUser) {
        boolean succesStatus = false;
        if (userRepository.findByEmail(newUser.getEmail()) == null) {
            userRepository.save(newUser);
            succesStatus = true;
        }

        return succesStatus;
    }
```
This method could also be written using a more functional approach with Optional.ofNullable (but let's leave it out of scope for the moment).